openml/extensions/sklearn_extension.py

```diff
@@ -0,0 +1,467 @@
+                           model_classes_mapping):
+        """Util function that turns probability estimates of a classifier for a
+        given instance into the right arff format to upload to openml.
+
+            Parameters
+            ----------
+            rep_no : int
+                The repeat of the experiment (0-based; in case of 1 time CV,
+                always 0)
+            fold_no : int
+                The fold nr of the experiment (0-based; in case of holdout,
+                always 0)
+            sample_no : int
+                In case of learning curves, the index of the subsample (0-based;
+                in case of no learning curve, always 0)
+            row_id : int
+                row id in the initial dataset
+            correct_label : str
+                original label of the instance
+            predicted_label : str
+                the label that was predicted
+            predicted_probabilities : array (size=num_classes)
+                probabilities per class
+            class_labels : array (size=num_classes)
+            model_classes_mapping : list
+                A list of classes the model produced.
+                Obtained by BaseEstimator.classes_
+
+            Returns
+            -------
+            arff_line : list
+                representation of the current prediction in OpenML format
+            """
+        if not isinstance(rep_no, (int, np.integer)):
+            raise ValueError('rep_no should be int')
+        if not isinstance(fold_no, (int, np.integer)):
+            raise ValueError('fold_no should be int')
+        if not isinstance(sample_no, (int, np.integer)):
+            raise ValueError('sample_no should be int')
+        if not isinstance(row_id, (int, np.integer)):
+            raise ValueError('row_id should be int')
+        if not len(predicted_probabilities) == len(model_classes_mapping):
+            raise ValueError('len(predicted_probabilities) != len(class_labels)')
+
+        arff_line = [rep_no, fold_no, sample_no, row_id]
+        for class_label_idx in range(len(class_labels)):
+            if class_label_idx in model_classes_mapping:
+                index = np.where(model_classes_mapping == class_label_idx)[0][0]
+                # TODO: WHY IS THIS 2D???
+                arff_line.append(predicted_probabilities[index])
+            else:
+                arff_line.append(0.0)
+
+        arff_line.append(class_labels[predicted_label])
+        arff_line.append(correct_label)
+        return arff_line
+
+    def is_hpo_class(self, model):
+        return isinstance(model, sklearn.model_selection._search.BaseSearchCV)
+
+    def assert_hpo_class(self, model):
+        if not isinstance(model, sklearn.model_selection._search.BaseSearchCV):
+            raise ValueError(
+                'Flow model is not an instance of sklearn.model_selection._search.BaseSearchCV'
+            )
```

Comment:
Still not sure if `assert_hpo_class` should have been non-abstract. I now see that custom error message, which is nice, but we also see that you e.g. now also stopped re-using code from `is_hpo_class` (whereas the base class does this well). It might be that I don't entirely understand what's going on, though.

Reply:
I will update the child class to use the `is_hpo_class` method, but will continue to provide a custom error messgae. Nevertheless, the method should not be static.